kirin/abstract_sncf_resource.py

```diff
@@ -49,45 +51,51 @@ def __init__(self, nav_wrapper, timeout, contributor, builder):
-            else:
-                set_rtu_status_ko(rt_update, e.message, is_reprocess_same_data_allowed=True)
-                model.db.session.add(rt_update)
-                model.db.session.commit()
-            record_call("failure", reason=six.text_type(e), contributor=self.contributor)
-            raise
-
-        try:
             # assuming UTF-8 encoding for all input
             rt_update.raw_data = rt_update.raw_data.encode("utf-8")
 
             # raw_input is interpreted
             trip_updates = self.builder(self.navitia_wrapper, self.contributor).build(rt_update)
-            _, log_dict = core.handle(rt_update, trip_updates, self.contributor, is_new_complete=is_new_complete)
-            record_call("OK", contributor=self.contributor)
-        except KirinException as e:
-            allow_reprocess = not isinstance(e, InvalidArguments)  # reprocess is useless if input is invalid
-            set_rtu_status_ko(rt_update, e.data["error"], is_reprocess_same_data_allowed=allow_reprocess)
-            model.db.session.add(rt_update)
-            model.db.session.commit()
-            record_call("failure", reason=six.text_type(e), contributor=self.contributor)
-            raise
+            _, handler_log_dict = core.handle(
+                rt_update, trip_updates, self.contributor, is_new_complete=is_new_complete
+            )
+            log_dict.update(handler_log_dict)
+
         except Exception as e:
-            set_rtu_status_ko(rt_update, e.message, is_reprocess_same_data_allowed=True)
-            model.db.session.add(rt_update)
-            model.db.session.commit()
-            record_call("failure", reason=six.text_type(e), contributor=self.contributor)
-            raise
+            status = "failure"
+            allow_reprocess = True
+            if is_invalid_input_exception(e):
+                status = "warning"  # Kirin did his job correctly if the input is invalid and rejected
+                allow_reprocess = False  # reprocess is useless if input is invalid
+
+            if rt_update is not None:
+                error = e.data["error"] if (isinstance(e, KirinException) and "error" in e.data) else e.message
+                set_rtu_status_ko(rt_update, error, is_reprocess_same_data_allowed=allow_reprocess)
+                model.db.session.add(rt_update)
+                model.db.session.commit()
+            else:
+                # rt_update is not built, make sure reprocess is allowed
+                allow_reprocess_same_data(self.contributor)
+
+            log_dict.update({"exc_summary": six.text_type(e), "reason": e})
 
-        duration = (datetime.utcnow() - start_datetime).total_seconds()
-        log_dict.update({"duration": duration})
-        record_call("Simple feed publication", **log_dict)
-        logging.getLogger(__name__).info("Simple feed publication", extra=log_dict)
+            record_custom_parameter("reason", e)  # use __str__() to have complete details
```

Comment:
Confusing comment. Does it mean that it doesn't record everything as is and if I want more details, I need to change the code?

Reply:
OK I understand the confusion.
This simple `e` is using __str__, and the result is not the same as six.text_type for KirinExceptions that overloads the method (using https://github.com/CanalTP/kirin/blob/5fa93fc2b001f0ed7d7469d028f1c12521b2f9a0/kirin/exceptions.py#L45-L54).

Suggesting:
```suggestion
            record_custom_parameter("reason", e)  # using __str__() here to have complete details
```